packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_client.dart

```diff
@@ -65,13 +73,36 @@ class EventClient implements Closeable {
   static final AmplifyLogger _logger =
       AmplifyLogger.category(Category.analytics).createChild('EventClient');
 
-  /// Received events are NEVER sent immediately but cached to be sent in a batch
-  Future<void> recordEvent(Event event) async {
-    await _storageAdapter.saveEvent(event);
+  /// Record event to be sent in the next event batch on [flushEvents]
+  Future<void> recordEvent({
+    required String eventType,
+    Session? session,
+    AnalyticsProperties? properties,
+  }) async {
+    final pinpointEvent = _eventCreator.createPinpointEvent(
+      eventType,
+      session,
+      properties,
+    );
+    await _storageAdapter.saveEvent(pinpointEvent);
+  }
+
+  /// Register [AnalyticsProperties] to be sent with all future events
+  Future<void> registerGlobalProperties(
```

Comment:
This is unnecessary abstraction. Construct the global properties manager at this level and inject into the event creator. The event creator should be single-purpose and just create events.

Reply:
I don't see how this is unnecessary abstraction.  Managing the metadata that is sent with events seems related to EventCreator functionality.  

Furthermore, switching to this requires changing the EventClient constructor call to use a factory instead because the `EndpointGlobalFieldsManager` instance must be the same.  

This extra complexity doesn't seem worth it and seems more unnecessary.

Reply:
Also, the `EventCreator` uses the `EventGlobalFieldsManager` to manage its own behavior when sending events. 

It seems weird to expose an inner class (`EventGlobalFieldsManager`) that will have side effects on `EventCreator`.  

Especially when `EventCreator` is the only class that needs to use `EventGlobalFieldsManager`.